.travis.yml

```diff
@@ -15,6 +15,13 @@
 #
 matrix:
   include:
+    - name: "Google java format validation"
+      # Validate with Google Java Style
+      language: java
+      before_script:
+        - wget https://github.com/google/google-java-format/releases/download/google-java-format-1.7/google-java-format-1.7-all-deps.jar
+      script:
+        - find . -iname "*.java" | xargs java -jar google-java-format-1.7-all-deps.jar --dry-run --set-exit-if-changed
```

Comment:
It looks like this is currently failing travis?

`The command "find . -iname "*.java" | xargs java -jar google-java-format-1.7-all-deps.jar --dry-run --set-exit-if-changed" exited with 123.`

Can you verify that the --dry-run doesn't fail the build? I imagine a lot of our files will probably fail the google-java-formatter.

Reply:
@seehamrun I might have misunderstood the issue.
Let me clarify 🙇 So, we need to add a step to check for google java format when someone makes a PR, right? 

**What I did**
`--dry-run` would just show a list of files which are not following the format. `--set-exit-if-changed` along with dry run option would exit with non-zero status code so that this can be caught during CI Check.

> It looks like this is currently failing Travis?

That's expected, right? because almost every file in this repo is not following the format 😢

Reply:
Yes but we dont want to block development by this. Lets do it in 3 steps
- turn on but only in dry-run and with a 0 status code so travis passes
- fix all of the files and show that dry-run returned no changes 
- update it so that a non-zero code is returned if there are violating files.

Reply:
Understood!

Reply:
One intermediate step we could do is only fail on the files that have been added or modified. Something like: `git diff --name-only --diff-filter=AM HEAD...$TRAVIS_BRANCH` and then filtering to just .java files. That way we can slowly migrate to the correct formatting and the blame is slightly less affected.